### PR TITLE
Making pending deployments asynchronous

### PR DESCRIPTION
**Pull request by narehayrapetyan**
#### Summary

Moving pending deployments to be async

#### Testing

- [x] daemon-scheduler binary built locally and unit-tests pass (`cd daemon-scheduler; make; cd ../`)
- [x] daemon-scheduler build in Docker succeeds (`cd daemon-scheduler; make release; cd ../`)

New tests cover the changes: yes

#### Licensing

This contribution is under the terms of the Apache 2.0 License: Yes

#### Before merging

- [x] daemon-scheduler integration tests pass. Required setup details are listed [here](https://github.com/blox/blox/blob/dev/daemon-scheduler/internal/features/README.md).
- [x] daemon-scheduler end-to-end tests pass. Required setup details are listed [here](https://github.com/blox/blox/blob/dev/daemon-scheduler/internal/features/README.md).

### CODE REVIEW

File: daemon-scheduler/pkg/deployment/deployment.go [hunk 1]
```diff
@@ -151,40 +156,54 @@ func (d deployment) CreateSubDeployment(ctx context.Context, environmentName str
...
 	// TODO: Figure out how we want to track failures in sub-deployments
 	if deployment.Status == types.DeploymentCompleted {
 		return deployment, nil
 	}
 
-	failures := resp.Failures
-	if deployment.FailedInstances != nil {
-		failures = append(failures, deployment.FailedInstances...)
+	if deployment.Status == types.DeploymentPending {
+		err = deployment.UpdateDeploymentInProgress(len(instanceARNs), failures)
```
Comment (kylbarnes) [inline daemon-scheduler/pkg/deployment/deployment.go]: Shouldn't this method be called `UpdateDeploymentToInProgress`? I read this as updating a deployment that was already in progress before I jumped into the method to see what it did.

File: daemon-scheduler/pkg/deployment/deployment_worker.go [hunk 2]
```diff
@@ -148,16 +201,20 @@ func (d deploymentWorker) updateDeploymentObject(deployment *types.Deployment,
 	resp *ecs.DescribeTasksOutput) (*types.Deployment, error) {
 
 	if d.deploymentCompleted(resp.Tasks, resp.Failures) {
-		return deployment.UpdateDeploymentCompleted(resp.Failures)
+		err := deployment.UpdateDeploymentCompleted(resp.Failures)
```
Comment (kylbarnes) [inline daemon-scheduler/pkg/deployment/deployment_worker.go]: Same comment. Shouldn't this be `UpdateDeploymentToCompleted`?

**Comment by kylbarnes** [comment]
Looks good to me. Only a couple comments regarding naming, which you're free to ignore. I was able to verify this functionality by running DS locally in debug mode and following along with the asynchronous steps.

**Comment by narehayrapetyan** [comment]
@kylbarnes renamed it. Can you approve unless you have other comments? Otherwise I can't merge.

**Pull request merged**